universityofadelaide/shepherd
# Feature/dynamic service routing

Introduce basic A/B style deployments for shepherd.

To test:
1. Create a new site with a 'prod' url. Shepherd will now create a service and route for that site - check in openshift.
2. Create a some new dev environments for that site.
3. Check in openshift, the original site service and route will be there as well as new ones for the dev environment. The dev one will work, the production one wont yet work.
4. On the environments page, from the drop-down, promote one of the environments
5. Both endpoints will now be associated with the environment selected.
6. Check in openshift, you can see the services have been updated.
7. Promote the other environment, it will switch back.

Also implemented in this MR
url on the site page
taxonomy changes to support the promotion/demotion
taxonomy changes to stop string comparing and use fields
